manifests/repo.pp

```diff
@@ -18,23 +18,11 @@
     descr    => "Pulpcore ${version}",
     baseurl  => pick($baseurl, "https://yum.theforeman.org/pulpcore/${version}/${dist_tag}/\$basearch"),
     enabled  => '1',
-    gpgcheck => if $gpgkey == '' { '0' } else { '1' },
+    gpgcheck => if $gpgkey == '' or $version == 'nightly' { '0' } else { '1' },
     gpgkey   => pick($gpgkey, "https://yum.theforeman.org/pulpcore/${version}/GPG-RPM-KEY-pulpcore"),
     notify   => Anchor['pulpcore::repo'],
   }
 
-  # Only EL8 has DNF modules
```

Comment:
Removing this here, leaves a "failsafe" module in DNF, but seems not hinder the upgrade:

```console
[root@centos8-stream ~]# dnf module list |grep -B5 pulpcore
Last metadata expiration check: 0:04:07 ago on Thu 23 Nov 2023 07:26:27 AM UTC.
@modulefailsafe
Name                 Stream          Profiles                                 Summary                                                                                                                                                                                                                                   
pulpcore             el8 [e]                                                  Pulpcore module                                                                                                                                          
```

The only cleanup needed is to remove the `/etc/dnf/modules.d/pulpcore.module` file, which I think can be done in here?

(People who do not use `pulpcore::repo` still need to be informed via docs, but I think that's fine)

Reply:
> (People who do not use `pulpcore::repo` still need to be informed via docs to either delete that file or call `dnf module reset pulpcore`, but I think that's fine)

https://github.com/theforeman/foreman-documentation/pull/2597